### PR TITLE
ARROW-4228: [GLib] Add garrow_list_data_type_get_field()

### PR DESCRIPTION
**Pull request by shiro615**
Because we use `get_field()` to get `GArrowField` in composite data type.
`garrow_list_data_type_get_value_field()` is deprecated.

### CODE REVIEW

File: c_glib/doc/arrow-glib/arrow-glib-docs.xml [hunk 1]
```diff
@@ -163,6 +163,10 @@
     <title>Index of deprecated API</title>
     <xi:include href="xml/api-index-deprecated.xml"><xi:fallback /></xi:include>
   </index>
+  <index id="api-index-0-13-0" role="0.13.0">
+    <title>Index of new symbols in 0.13.0</title>
+    <xi:include href="xml/api-index-0.12.0.xml"><xi:fallback /></xi:include>
```
Comment (kou) [inline c_glib/doc/arrow-glib/arrow-glib-docs.xml]: Could you replace the "0.12.0" with "0.13.0"?

**Review by kou — CHANGES_REQUESTED** [review]
Thanks!
Could you check my comment?

File: c_glib/doc/arrow-glib/arrow-glib-docs.xml (hunk 1 above)
Comment (shiro615) [reply c_glib/doc/arrow-glib/arrow-glib-docs.xml]: Thank you for pointing out it. I've fixed it.

**Review by kou — APPROVED** [review]
+1
I'll merge this.

**Pull request closed without merging**